README.md

```diff
@@ -1,2 +1,41 @@
 # comp110-worksheet-5
 Base repository for COMP110 worksheet 5
+
+(a)
+
+this algorithm compares each item of the i and j lists and checks if their values are the same. After comparing two values, it will either keep comparing the next two until the list is finished, where it will return false and end the loop, or if the algorithm detects if two values are the same, it will return true and end the loop.
+
+(b)
+Because a for loop is being run within a for loop, this creates a nested loop and therefore a quadratic running time.
+
+(c)
+This algorithm would still be functional, as n represents the same length for i and j. Because i and j have the same total length,  i can be used to represent the length of j.
+
+
+(d)
+Doing this would allow the program to only have to calculate the value of one list. Processing the length of both lists would only need to be done once by i, and j can simply reference the result.
```

Comment:
I think you have the right idea -- if by "list" you mean the set of pairs of elements in the list